src/spesh/inline.c

```diff
@@ -1056,6 +1056,23 @@ static void return_to_box(MVMThreadContext *tc, MVMSpeshGraph *g, MVMSpeshBB *re
     MVM_spesh_manipulate_release_temp_reg(tc, g, type_temp);
 }
 
+static void return_to_unbox(MVMThreadContext *tc, MVMSpeshGraph *g, MVMSpeshBB *return_bb,
+        MVMSpeshIns *return_ins, MVMSpeshOperand target, MVMuint16 unbox_op) {
+    MVMuint16         register_type = unbox_op == MVM_OP_unbox_i ? MVM_reg_int64 :
+                                      unbox_op == MVM_OP_unbox_u ? MVM_reg_int64 :
+                                      unbox_op == MVM_OP_unbox_n ? MVM_reg_num64 :
+                                                                   MVM_reg_str;
+    /* Replace the return with an unboxing instruction. */
+    MVMSpeshOperand ver_target      = MVM_spesh_manipulate_new_version(tc, g, target.reg.orig);
+    MVMSpeshOperand *unbox_operands = MVM_spesh_alloc(tc, g, 2 * sizeof(MVMSpeshOperand));
+    return_ins->info                = MVM_op_get_op(unbox_op);
+    return_ins->operands            = unbox_operands;
+    unbox_operands[0]               = ver_target;
+    unbox_operands[1]               = return_ins->operands[0];
```

Comment:
Hm, this looks wrong. I think we're meant to update `return_ins->operands = unbox_operands` *after* the population of `unbox_operands`.

Reply:
Well, in happens in this order in `return_to_box` https://github.com/MoarVM/MoarVM/pull/1524/files#diff-b4fbc9fb84a8bb3a3b9d3371048903867537c428ae7be8a781d92574bbfb84c5R1036-R1039

Reply:
Yes, but it's being stuck on a freshly allocated `MVMSpeshIns` there; here we're updating the one instruction.

Reply:
Ah, right. However, just looking at `return_to_set` made me think maybe should we instead have a `return_to_op`? Then `return_to_set` and `return_to_unbox` (which already takes an op as an argument) can be combined.

Reply:
Then the diff would look like this https://gist.github.com/MasterDuke17/a5842d1e8fa5a45cafc7c1b5d8d07a38

Reply:
@MasterDuke17 Yes, that's a good way to do it. :+1: